### PR TITLE
Turn off masking

### PR DESCRIPTION
**Pull request by profxj**
Mainly propagates masking choice into collate.

No new docs or tests

### CODE REVIEW

File: linetools/spectra/utils.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def splice_two(spec1, spec2, wvmx=None, scale=1., chk_units=True):
     return spec3
 
 
-def collate(spectra):
+def collate(spectra, **kwargs):
     """ Generate a single XSpectrum1D instance containing an array of
```
Comment (ntejos) [inline linetools/spectra/utils.py]: Please add in docstring where are the `**kwargs` passed to.

File: linetools/spectra/xspectrum1d.py [hunk 2]
```diff
@@ -766,13 +767,19 @@ def rebin(self, new_wv, all=False, **kwargs):
         all : bool, optional
           Rebin all spectra in the XSpectrum1D object?
           Set masking='none' to have the resultant spectra all be regsitered, but note
-             that there will still be masking
+             that there will still be masking unless otherwise specified in kwargs
```
Comment (ntejos) [inline linetools/spectra/xspectrum1d.py]: I did not understand the comment regarding setting masking to 'none'. What is the idea here?

Also, check typo `regsitered`.

File: linetools/spectra/xspectrum1d.py [hunk 3]
```diff
@@ -766,13 +767,19 @@ def rebin(self, new_wv, all=False, **kwargs):
...
         -------
         XSpectrum1D of the rebinned spectrum
         """
         from .utils import rebin, collate
+        #
+        if 'masking' in kwargs:
+            masking = kwargs['masking']
+        else:
+            masking = 'edges'
+        #
```
Comment (ntejos) [inline linetools/spectra/xspectrum1d.py]: Is this necessary? Given the default masking for `XSpectrum1D` (and hence `collate()`), is `masking='edges'` anyways...

File: linetools/spectra/utils.py (hunk 1 above)
Comment (profxj) [inline linetools/spectra/utils.py]: done

File: linetools/spectra/xspectrum1d.py (hunk 3 above)
Comment (profxj) [inline linetools/spectra/xspectrum1d.py]: Now set to none here and in the __init__

This is a somewhat big change, but is the way 
I should have had it and it is likely the way we
will be using the new Spectrum1D object.
 
I hope it doesn't cause trouble to existing scripts,
but there are likely to be some repercussions..

**Comment by profxj** [comment]
merging

**Pull request merged**